staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -287,9 +288,11 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			// If that's the case wait and resend watch request.
 			if urlError, ok := err.(*url.Error); ok {
 				if opError, ok := urlError.Err.(*net.OpError); ok {
-					if errno, ok := opError.Err.(syscall.Errno); ok && errno == syscall.ECONNREFUSED {
-						time.Sleep(time.Second)
-						continue
+					if osSyscallError, ok := opError.Err.(*os.SyscallError); ok {
```

Comment:
this conditional nested casting of error types is fragile to changes in the structure of the returned error. suggest adding an `IsConnectionRefused` method to `k8s.io/apimachinery/pkg/util/net/util.go` that matches the implementation of `IsConnectionReset` (but checks `ECONNREFUSED`) instead.

a unit test specifically around connection-refused detection can be added to that package

Reply:
Thank you for your suggestion, I have already added the `IsConnectionRefused` method and its unit test, please review it again, thank you every much.